src/mapml.js

```diff
@@ -3270,16 +3595,143 @@ M.MapMLFeatures = L.FeatureGroup.extend({
       // info: https://github.com/Leaflet/Leaflet/pull/4597 
       L.DomUtil.addClass(this._container,'leaflet-pane mapml-vector-container');
       L.setOptions(this.options.renderer, {pane: this._container});
-
       this._layers = {};
-
       if (mapml) {
+        //needed to check if the feature is static or not, since this method is used by templated also
+        if(!mapml.querySelector('extent') && mapml.querySelector('feature')){
+          this._features = {};
+          this._staticFeature = true;
+          this.isVisible = true; //placeholder for when this actually gets updated in the future
+          this.zoomBounds = this._getZoomBounds(mapml);
+          this.layerBounds = this._getLayerBounds(mapml);
+          L.extend(this.options, this.zoomBounds);
+        }
         this.addData(mapml);
+        if(this._staticFeature){
+          this._resetFeatures(this._clampZoom(this.options._leafletLayer._map.getZoom()));
+
+          this.options._leafletLayer._map._addZoomLimit(this);
+        }
       }
     },
-    
+
     getEvents: function(){
-      return {'moveend':this._removeCSS};
+      if(this._staticFeature){
+        return {
+          'moveend':this._handleMoveEnd,
+        };
+      }
+      return {
+        'moveend':this._removeCSS
+      };
+    },
+
+    _handleMoveEnd : function(){
+      let zoomLevel = this._map.getZoom();
+      zoomLevel = zoomLevel > this.zoomBounds.maxNativeZoom? this.zoomBounds.maxNativeZoom: zoomLevel;
+      zoomLevel = zoomLevel < this.zoomBounds.minNativeZoom? this.zoomBounds.minNativeZoom: zoomLevel;
+      this.isVisible = this._checkZoom() && this._layers && this.layerBounds && 
+                        this.layerBounds.overlaps(
+                          M.pixelToPCRSBounds(
+                            this._map.getPixelBounds(),
+                            this._map.getZoom(),this._map.options.projection));
+      this._removeCSS();
```

Comment:
Why do we need to remove css on moveend?

Reply:
That is related to an issue from a few months ago, where we needed to garbage collect styles and stylesheets. It used to be the moveend handler but now since static features get assigned a different moveend handler the old handler still needs to be invoked to garbage collect the styles and stylesheets if there are any.

Reply:
ok